core/src/main/scala/org/apache/spark/h2o/backends/SharedBackendUtils.scala

```diff
@@ -113,6 +113,10 @@ private[backends] trait SharedBackendUtils extends Logging with Serializable {
       conf.setH2OClientLogLevel(incLogLevel(conf.h2oClientLogLevel, "INFO"))
     }
 
+    if(!conf.hadoopConf.isDefined){
```

Comment:
I would apply the automatic hadoop configuration just in case of Internal backend for now. It is fine to just move this piece of code to to https://github.com/h2oai/sparkling-water/blob/ad063d605e976d6770730c41024170dcae73a0e7/core/src/main/scala/org/apache/spark/h2o/backends/internal/InternalH2OBackend.scala#L59